lib/jekyll-titles-from-headings/generator.rb

```diff
@@ -2,9 +2,13 @@ module JekyllTitlesFromHeadings
   class Generator < Jekyll::Generator
     attr_accessor :site
 
-    TITLE_REGEX = %r!\A\s*\#{1,3} (.*)\n$!
+    TITLE_REGEX = %r!\A\s*\#{1,3}\s+(.*)\n$!
     CONVERTER_CLASS = Jekyll::Converters::Markdown
 
+    # Regex to strip extra markup still present after markdownify
+    # (footnotes at the moment).
+    EXTRA_MARKUP_STRIP = %r!\[\^[^]]*\]!
```

Comment:
I’m not sure what is allowed, exactly, as footnote id (are underscores allowed, e.g.?). With that, I just made it accept `[^whatever is there]`.

If you prefer the more stringent regex, please let me know and I’ll change it no prob.